adapters/VerizonMedia/VerizonMediaAdapter/GADMVerizonAdapterBaseClass.m

```diff
@@ -0,0 +1,593 @@
+    }
+    self.interstitialAd = nil;
+    self.interstitialAdFactory = [[VASInterstitialAdFactory alloc] initWithPlacementId:self.placementID vasAds:self.vasAds delegate:self];
+    [self.rewardConnector adapterDidSetUpRewardBasedVideoAd:self];
+}
+
+#pragma mark - common
+
+- (BOOL)prepareAdapterForAdRequest {
+    if (!self.placementID || ! self.vasAds.isInitialized) {
+        NSError *error = [NSError errorWithDomain:kGADErrorDomain
+                                             code:kGADErrorMediationAdapterError
+                                         userInfo:@{NSLocalizedDescriptionKey : @"Verizon adapter not properly intialized."}];
+        [self.gadConnector adapter:self didFailAd:error];
+        [self.rewardConnector adapter:self didFailToSetUpRewardBasedVideoAdWithError:error];
+        return NO;
+    }
+    
+    [self setRequestInfoFromConnector:self.connector];
+    
+    return YES;
+}
+
+- (void)stopBeingDelegate
+{
+    if (self.inlineAd) {
+        if ([self.inlineAd respondsToSelector:@selector(destroy)]) {
+            [self.inlineAd performSelector:@selector(destroy)];
+        } else {
+            NSLog(@"GADMAdapterVerizon: The adapter is intended to work with Verizon Ads SDK version 1.0.4 or higher.  Please update the Verizon Ads SDK.");
+        }
+    }
+    
+    if (self.interstitialAd) {
+        if ([self.interstitialAd respondsToSelector:@selector(destroy)]) {
+            [self.interstitialAd performSelector:@selector(destroy)];
+        } else {
+            NSLog(@"GADMAdapterVerizon: The adapter is intended to work with Verizon Ads SDK version 1.0.4 or higher.  Please update the Verizon Ads SDK.");
+        }
+    }
+    
+    self.inlineAdFactory.delegate = nil;
+    self.inlineAd.delegate = nil;
+    self.interstitialAdFactory.delegate = nil;
+    self.interstitialAd.delegate = nil;
+    self.inlineAd = nil;
+    self.interstitialAd = nil;
+}
+
+#pragma mark - private
+
+- (void)setRequestInfoFromConnector:(id<GADMediationAdRequest>)connector
+{
+    //User Settings
+    [self setUserSettingsFromConnector:connector];
+    
+    //COPPA
+    [self setCoppaFromConnector:connector];
+    
+    // Location
+    if (connector.userHasLocation) {
+        self.vasAds.locationEnabled = YES;
+    }
+}
+
+- (void)setUserSettingsFromConnector:(id<GADMediationAdRequest>)connector
+{
+    VASRequestMetadataBuilder *builder = [[VASRequestMetadataBuilder alloc] init];
+    
+    // Mediator
+    builder.appMediator = [NSString stringWithFormat:@"AdMobVAS-%@",  [GADMVerizonAdapterBaseClass adapterVersion]];
+    
+    // Keywords
+    if ([connector userKeywords] != nil && [[connector userKeywords] count] > 0) {
+        builder.userKeywords = [connector userKeywords];;
+    }
+    
+    self.vasAds.requestMetadata = [builder build];
+}
+
+- (void)setCoppaFromConnector:(id<GADMediationAdRequest>)connector
+{
+    self.vasAds.COPPA = [connector childDirectedTreatment];
+}
+
+- (CGRect)makeBannerRect:(GADAdSize)adSize
```

Comment:
Why did we remove the adaptive banner code and added `makeBannerRect: ` method?

Reply:
I believe this is expected behavior as Google Mobile Ads SDK checks if the available sizes satisfies the min fill ratio required to accept the size. The supported size width should fill more than 50% of the requested size width and the supported size height should fill more than 70% of the requested size height. If the supported ad sizes doesn't satisfy the above conditions we send with size -1, -1. So in the example cases you provided:
* Requested ad size: [width: 729, height: 50], banner size of [width: 320, height: 50] is not expected as it doesn't fill 50% of the requested width hence got [width: -1, height: -1].
* Second case works fine.
* Requested ad size: [width: 600, height: 300], medium rectangle size of [width: 300, height: 250] doesn't fill more than 50% of the requested width hence got [width: -1, height: -1].

So please revert back to the code we have added i.e, `GADClosestValidSizeForAdSizes`.

Reply:
Thank you for the clarifications @maninomula!